### ISSUE TITLE
Date displays should be independant of actual dates

### ISSUE THREAD

**Issue by kalepail** [Baremetrics/calendar#29]
i.e. If you format the input dates to just show months (not sure why you'd want to do that but still) the switcher buttons will show "Invalid Date" because they look at the html of the inputs when they really should be looking at actual date objects or strings independant of what's being displayed to the end user.

**Comment by SleeplessByte** [comment]
Yeah, you might want to either store it internally, or use data objects just like the month and year storage.

On my own version I store all the dates as `.data( 'unix', somemomentobject.unix() )` and parse it back in with `moment( ...data('unix' ) )`

**Comment by kalepail** [comment]
Mhm. A project for another day but it's definitely something that needs doing.
